### PR TITLE
Pe cores configuration testing

### PR DESCRIPTION
**Pull request by JanPrusinowski**

### CODE REVIEW

File: dasharo-compatibility/cpu-cores-count.robot [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+...                     Skip If    not ${CPU_TESTS_SUPPORT}    CPU tests not supported
+Suite Teardown      Run Keyword
+...                     Log Out And Close Connection
+
+
+*** Test Cases ***
+CCC001.001 CPU Hyper-Threading disabled (Ubuntu)
+    [Documentation]    Check if Hyper-Threading is disabled.
+    Set UEFI Option    HyperThreading    ${FALSE}
+    Power On
```
Comment (macpijan) [inline dasharo-compatibility/cpu-cores-count.robot:33]: @JanPrusinowski Power On seems redundant, the `Set UEFI Option` kwd does the platform reset

File: dasharo-compatibility/cpu-cores-count.robot [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+    IF    '${threads[:1]}' == '1' 
+        ${count}=    Set Variable    1
+    ELSE IF    '${threads[:1]}' == '2'
+        ${count}=    Set Variable    2 
+    ELSE
+        Fail    Hyper-Threading status could not be established.
+    END
+    RETURN     ${count}
+
+Get Last Word From String
```
Comment (macpijan) [inline dasharo-compatibility/cpu-cores-count.robot:224]: The proposed keyword seems too complex.

Would something like this (or similar) work?

```
    ${words}=  Split String  ${string}  ${SPACE}
    ${last_word}=  Get From List  ${words}  -1
```

Alternatively, if you want to use python anyway, you can create a Python function in a .py file. But RF should be enough here.

Comment (JanPrusinowski) [reply dasharo-compatibility/cpu-cores-count.robot]: Changed in: 58b37a0a3e56b4f758ad352d5db6046131e7ece2

**Comment by macpijan** [comment]
Example results on VP6670:

```
==============================================================================
Cpu-Cores-Count                                                               
==============================================================================
CCC001.001 Check core count with HT disabled (Ubuntu) :: Disable H... .....
Checking if util-linux is installed...

Package util-linux is installed
CCC001.001 Check core count with HT disabled (Ubuntu) :: Disable H... | PASS |
------------------------------------------------------------------------------
CCC002.001 Check core count with HT enabled :: Enable HT and check... .....
Checking if util-linux is installed...

Package util-linux is installed
CCC002.001 Check core count with HT enabled :: Enable HT and check... | PASS |
------------------------------------------------------------------------------
CCC003.001 Check core count (HT Enabled, P: All, E: 0) (Ubuntu)       .       
Checking if cpuid is installed...

Package cpuid is installed
CCC003.001 Check core count (HT Enabled, P: All, E: 0) (Ubuntu)       | PASS |
------------------------------------------------------------------------------
CCC004.001 Check core count (HT Enabled, P: All, E: All) (Ubuntu)     .       
Checking if cpuid is installed...

Package cpuid is installed
CCC004.001 Check core count (HT Enabled, P: All, E: All) (Ubuntu)     | PASS |
------------------------------------------------------------------------------
CCC005.001 Check core count (HT Disabled, P: All, E: 0) (Ubuntu)      ..      
Checking if cpuid is installed...

Package cpuid is installed
CCC005.001 Check core count (HT Disabled, P: All, E: 0) (Ubuntu)      | PASS |
------------------------------------------------------------------------------
CCC006.001 Check core count (HT Disabled, P: All, E: 0) (Ubuntu)      ..      
Checking if cpuid is installed...

Package cpuid is installed
CCC006.001 Check core count (HT Disabled, P: All, E: 0) (Ubuntu)      | PASS |
------------------------------------------------------------------------------
CCC007.001 Check core count (HT Enabled, P: 1, E: A) (Ubuntu)         .       
Checking if cpuid is installed...

Package cpuid is installed
CCC007.001 Check core count (HT Enabled, P: 1, E: A) (Ubuntu)         | PASS |
------------------------------------------------------------------------------
CCC008.001 Check core count (HT Disabled, P: 1, E: A) (Ubuntu)        ..      
Checking if cpuid is installed...

Package cpuid is installed
CCC008.001 Check core count (HT Disabled, P: 1, E: A) (Ubuntu)        | PASS |
------------------------------------------------------------------------------
CCC009.001 Check core count (HT Enabled, P: 1, E: 1) (Ubuntu)         ...     
Checking if cpuid is installed...

Package cpuid is installed
CCC009.001 Check core count (HT Enabled, P: 1, E: 1) (Ubuntu)         | PASS |
------------------------------------------------------------------------------
CCC010.001 Check core count (HT Disabled, P: 1, E: 1) (Ubuntu)        ...     
Checking if cpuid is installed...

Package cpuid is installed
CCC010.001 Check core count (HT Disabled, P: 1, E: 1) (Ubuntu)        | PASS |
------------------------------------------------------------------------------
CCC011.001 Check core count (HT Enabled, P: A, E: 1) (Ubuntu)         ..      
Checking if cpuid is installed...

Package cpuid is installed
CCC011.001 Check core count (HT Enabled, P: A, E: 1) (Ubuntu)         | PASS |
------------------------------------------------------------------------------
CCC012.001 Check core count (HT Disabled, P: A, E: 1) (Ubuntu)        ..      
Checking if cpuid is installed...

Package cpuid is installed
CCC012.001 Check core count (HT Disabled, P: A, E: 1) (Ubuntu)        | PASS |
------------------------------------------------------------------------------
Cpu-Cores-Count                                                       | PASS |
12 tests, 12 passed, 0 failed
==============================================================================
```

**Pull request merged**